snap-packages/launchers/nbjavac-cleanup

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+# Remove externally installed nbjavac references from the userdir
+find ${SNAP_USER_DATA} -name '*nbjavac.*' -delete || true
```

Comment:
Is this because you're reusing user and cache folders?

Reply:
Cache folders are never re-used they are unique for every snap revision (and usually cleaned up by the janitor module later on). The user-dir is not really reused, but Snap creates a copy of the installed revision to a new revision, so if the user is not satisfied with the new revision he/she can safely revert back to the previously installed one.

Reply:
Ah, OK.  Glad cache is left out.  Is userdir copied between different versions of the IDE too?

Reply:
Yes. I think only if they are not exists. So when someone upgrades from 12.6 (rev: 52) to 13-rc3 (rev: 56), then the user home at ```$HOME/snap/netbeans/52``` is copied over to  ```$HOME/snap/netbeans/56``` automatically before the next start of the IDE. If the user decides, that the update was not successful, and whishes to go back to the previously installed version by using ```snap revert netbeans``` it won't copy anything, but ```$HOME/snap/netbeans/52``` would be used as USER_HOME again (and of course the old cache dir)
So this is an optimistic auto-upgrade strategy, which lets the user go back to a previously known good state if the upgraded version does not meet the expectations.